tuxemon/player.py

```diff
@@ -76,3 +77,31 @@ def update(self, time_delta: float) -> None:
         diff_y = abs(after.y - before.y)
 
         self.game_variables["steps"] += diff_x + diff_y
+        """
+        %H - Hour 00-23
+        %j - Day number of year 001-366
+        """
+        self.game_variables["hour"] = dt.datetime.now().strftime("%H")

```

Comment:
Save the result of datetime.now to a variable, as IIRC, it's a bit slow.

Check the FPS of the game (there is a config option for this) and make sure there isn't a performance hit.

Reply:
I checked FPS and I didn't notice a performance hit. I noticed that FPS increases only after enter or exit a location (around 70/80) for a couple of seconds, if not it stays around 60, 59, etc. I tried with and without my commit.

Reply:
extract into variable please int(dt.datetime.now().strftime("%H"))

Even without performance issue, it's more concise

Reply:
```
        var = self.game_variables
        var["hour"] = dt.datetime.now().strftime("%H")
        var["daynr"] = dt.datetime.now().strftime("%j")

        # Day and night basic cycle (12h cycle)
        if int(var["hour"]) < 6:
            self.game_variables["daytime"] = False
        elif 6 <= int(var["hour"]) < 18:
            self.game_variables["daytime"] = True
        else:
            self.game_variables["daytime"] = False
```
You mean something like this? Did I understand?